client/blockchain.go

```diff
@@ -107,6 +112,43 @@ func NewNetworkFromConfig(conf *config.Config) (BlockchainNetwork, error) {
 	)
 }
 
+// MultipleNetworks enables launching multiple networks for simultaneous usage in test scenarios
+func MultipleNetworks(networkIDs ...string) MultiNetworkInit {
+	return func(conf *config.Config) ([]BlockchainNetwork, error) {
+		networkTracker := make(map[string]bool)
+		networks := make([]BlockchainNetwork, len(networkIDs))
+		for index, networkID := range networkIDs {
+			// Cannot handle multiple of the same network, throws our templating all out of whack and quickly becomes a
```

Comment:
Is this a problem with templating since you're using `index` in the Chainlink deployments and the values assignment in the map for the network deployments doesn't matter if there's clobbering in the map?

We'll need to support multiple of the same network, as for CCIP the most basic test will be `ethereum_geth` and `ethereum_geth`.

Reply:
Yes to a problem with templating, no to it being with chainlink. It was primarily an issue of the templating for the networks themselves. Looking at this freshly though, and with some of the changes around templating strategy in general I made, I think I can give this another shot.

At first, I imagined that the `ethereum_geth` <-> `ethereum_geth` would be easily replaced by something like `ethereum_geth` <-> `ethereum_hardhat`, but now I'm remembering that hardhat and ganache have some block subscription issues.